src/main/java/edu/cornell/kfs/sys/service/impl/BatchFileDirectoryServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package edu.cornell.kfs.sys.service.impl;
+
+import edu.cornell.kfs.sys.service.BatchFileDirectoryService;
+import edu.cornell.kfs.sys.util.SubDirectoryWalker;
+import org.kuali.rice.core.api.util.KeyValue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchFileDirectoryServiceImpl implements BatchFileDirectoryService {
+
+    @Override
+    public List<KeyValue> buildDirectoryKeyValuesList(List<File> rootDirectories) {
```

Comment:
Would it make more sense to define the "Cacheable" on the implementation rather than on the interface?